core/internal/example/module.go

```diff
@@ -0,0 +1,95 @@
+package example
+
+import (
+	"context"
+
+	"github.com/gogo/protobuf/proto"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"cosmossdk.io/core/appmodule"
+	"cosmossdk.io/core/blockinfo"
+	"cosmossdk.io/core/event"
+	"cosmossdk.io/core/gas"
+	"cosmossdk.io/core/store"
+	"github.com/cosmos/cosmos-sdk/depinject"
+)
+
+func init() {
+	// register the module with the app-wiring dependency injection framework
+	appmodule.Register(&Module{},
+		appmodule.Provide(func(k keeper) *appmodule.Handler {
+			h := &appmodule.Handler{}
+			RegisterMsgServer(h, k)
+			RegisterQueryServer(h, k)
+			return h
+		}),
+	)
+}
+
+// the module's dependency injection inputs
+type keeper struct {
+	depinject.In
+
+	KVStoreKey       store.KVStoreService
```

Comment:
The runtime module would define providers for the various implementations of these services?

Reply:
Yes